pkg/util/provider/machinecodes/status/status.go

```diff
@@ -88,10 +107,18 @@ func FromError(err error) (s *Status, ok bool) {
 
 	if matches, errInFind := findInString(err.Error()); errInFind == nil {
 		code := codes.StringToCode(matches[0])
-		return New(code, matches[1]), true
+		return &Status{
+			code:    int32(code),
+			message: matches[1],
+			cause:   err,
+		}, true
 	}
 
-	return New(codes.Unknown, err.Error()), false
+	return &Status{
+		code:    int32(codes.Unknown),
+		message: err.Error(),
+		cause:   err,
+	}, false
```

Comment:
```suggestion
	return WrapError(codes.Unknown, err.Error(), err), false
```
Same as above

Reply:
Same reason as above. Let us keep it this way.